src/coap_io.c

```diff
@@ -935,7 +935,8 @@ coap_socket_send(coap_socket_t *sock, const coap_session_t *session,
       default:
         /* error */
         coap_log_warn("protocol not supported\n");
-        bytes_written = -1;
+        /* CID 15664879 */
```

Comment:
1566479

Reply:
Correct and separate commit pushed for review 1 (to be squashed later)